proto/wippersnapper/signal.proto

```diff
@@ -0,0 +1,104 @@
+    // servo.proto
+    wippersnapper.servo.ServoAdd servo_add       = 40;
+    wippersnapper.servo.ServoRemove servo_remove = 41;
+    wippersnapper.servo.ServoWrite servo_write   = 42;
+    // pwm.proto
+    wippersnapper.pwm.PWMAdd pwm_add                              = 50;
+    wippersnapper.pwm.PWMRemove pwm_remove                        = 51;
+    wippersnapper.pwm.PWMWriteDutyCycle pwm_write_duty            = 52;
+    wippersnapper.pwm.PWMWriteDutyCycleMulti pwm_write_duty_multi = 53;
+    wippersnapper.pwm.PWMWriteFrequency pwm_write_freq            = 54;
+    // pixels.proto
+    wippersnapper.pixels.PixelsAdd pixels_add       = 60;
+    wippersnapper.pixels.PixelsRemove pixels_remove = 61;
+    wippersnapper.pixels.PixelsWrite pixels_write   = 62;
+    // ds18x20.proto
+    wippersnapper.ds18x20.Ds18x20Add ds18x20_add       = 70;
+    wippersnapper.ds18x20.Ds18x20Remove ds18x20_remove = 71;
+    // uart.proto
+    wippersnapper.uart.UartAdd uart_add       = 80;
+    wippersnapper.uart.UartRemove uart_remove = 81;
+    wippersnapper.uart.UartWrite uart_write = 82;
+    // i2c.proto
+    wippersnapper.i2c.I2cBusScan i2c_bus_scan                      = 90;
+    wippersnapper.i2c.I2cDeviceAddOrReplace i2c_device_add_replace = 91;
+    wippersnapper.i2c.I2cDeviceRemove i2c_device_remove            = 92;
+    wippersnapper.i2c.I2cDeviceOutputWrite i2c_device_output_write = 93;
+    // error.proto
+    wippersnapper.error.Error error = 100;
+  }
+}
+
+/*
+ * DeviceToBroker
+ * The DeviceToBroker message is sent from the device to the broker.
+ * It contains a oneof payload, which is a union of all the possible
+ * messages that can be sent from a device to the broker.
+ */
+message DeviceToBroker {
+  option (nanopb_msgopt).submsg_callback = true;
+  oneof payload {
+    //digitalio.proto
+    wippersnapper.digitalio.DigitalIOEvent digitalio_event = 10;
+    // analogio.proto
+    wippersnapper.analogio.AnalogIOEvent analogio_event = 20;
+    // checkin.proto
+    wippersnapper.checkin.CheckinRequest checkin_request = 30;
+    // servo.proto
+    wippersnapper.servo.ServoAdded servo_added = 40;
+    // pwm.proto
+    wippersnapper.pwm.PWMAdded pwm_added = 50;
+    // pixels.proto
+    wippersnapper.pixels.PixelsAdded pixels_added = 60;
+    // ds18x20.proto
+    wippersnapper.ds18x20.Ds18x20Added ds18x20_added = 70;
+    wippersnapper.ds18x20.Ds18x20Event ds18x20_event = 80;
+    // uart.proto
+    wippersnapper.uart.UartAdded uart_added            = 90;
+    wippersnapper.uart.UartWritten uart_written        = 91;
+    wippersnapper.uart.UartInputEvent uart_input_event = 92;
+    // i2c.proto
+    wippersnapper.i2c.I2cBusScanned i2c_bus_scanned                      = 110;
+    wippersnapper.i2c.I2cDeviceAddedOrReplaced i2c_device_added_replaced = 111;
+    wippersnapper.i2c.I2cDeviceRemoved i2c_device_removed                = 112;
+    wippersnapper.i2c.I2cDeviceEvent i2c_device_event                    = 113;
```

Comment:
These should've been in the `100`s not the `110`s, so they (and later messages) should be shifted `-20` after the ds18x20 change above